dnase/parseSamtoolsGenotypesToBedFiles.pl

```diff
@@ -58,25 +63,35 @@
                 my ($refFwd, $refRev, $nonrefFwd, $nonrefRev) = split /,/, $value;
                 $numRef = $refFwd+$refRev;
                 $numNonRef = $nonrefFwd+$nonrefRev;
+            } elsif ("END" eq $key) {
+                # structural variant coords (i.e. Delly) require overriding the prior coordinates using END
+                $max1 = int($value) - 1;
+            } elsif ("PE" eq $key) {
+                $PE = int($value);
             }
         }
-        unless (defined($totalDP)) {
-            die "Failed to parse DP from INFO ($info) in $_\n";
+        if ($min_total_depth_threshold > 0 and ($totalDP ne undef and $totalDP < $min_total_depth_threshold)) {
+            #warn "Skipping $chrom $pos with total depth $totalDP < $min_total_depth_threshold\n";
+            next;
         }
-        #if ($totalDP < $min_total_depth_threshold) {
-        #    #warn "Skipping $chrom $pos with total depth $totalDP < $min_total_depth_threshold\n";
-        #    next;
-        #}
         if ($min_allele_depth_threshold > 0 and (($numRef ne "NA" and $numRef < $min_allele_depth_threshold) or ($numNonRef ne "NA" and $numNonRef < $min_allele_depth_threshold))) {
             #warn "Skipping $chrom $pos with ref/nonref depth $numRef/$numNonRef < $min_allele_depth_threshold\n";
             next;
         }
-        my $min0 = $pos - 1;
-        my $max1 = $pos - 1 + length($ref) ;
+        
         my $bed3 = "$chrom\t$min0\t$max1";
         for (my $i = 0; $i < $NUM_SAMPLES; ++$i) {
             my $sampleName = $sampleNames[$i];
             my ($betterGT, $GQ, $DP, $PL) = parseGenotype( $format, $genotypes[$i], \%code2base );
+            # For Delly use PE as the DP metric
+            # DP is undef for delly as PE is used instead
+            if (! defined($DP) && defined($PE)) {
+                $DP = $PE;
+            }
+            # By this point DP should be defined for both delly and bcftools
+            unless (defined($DP)) {
+                die "Failed to parse DP from ($format), PE=$PE\n";
+            }
```

Comment:
Mark, it looks like you are filling FORMAT/DP with INFO/PE here. In light of the other comment, I think this isn't the right way to go so I'm going to remove this.

Reply:
That is what I am doing. Removing that code will leave $readDepth/$DP as undefined for delly entries.

Reply:
I am not sure if that might cause an issue when trying to write the output to file?

Reply:
take a look after my latest commit, which sets DP="NA" but should tolerate it

Reply:
it should be ok if "NA" shows up in the output bed

Reply:
That should work